wgengine/router/ifconfig_windows.go

```diff
@@ -622,28 +604,51 @@ func unicastIPNets(ifc *winipcfg.IPAdapterAddresses) []*net.IPNet {
 // doing the minimum number of AddAddresses & DeleteAddress calls.
 // This avoids the full FlushAddresses.
 //
-// Any IPv6 link-local addresses are not deleted.
+// Any IPv6 link-local addresses are not deleted out of caution as some
+// configurations may repeatedly re-add them. Link-local addresses are adjusted
+// to apply SkipAsSource to prevent them from being used, a the addresses will
+// not be routed on the Tailscale interface.
 func syncAddresses(ifc *winipcfg.IPAdapterAddresses, want []*net.IPNet) error {
-	var erracc error
+	var erracc []error
 
 	got := unicastIPNets(ifc)
 	add, del := deltaNets(got, want)
-	del = excludeIPv6LinkLocal(del)
+
+	ll := make([]*net.IPNet, 0)
 	for _, a := range del {
+		// do not delete link-local addresses, and collect them for later
+		// applying SkipAsSource.
+		if isIPv6LinkLocal(a) {
+			ll = append(ll, a)
+			continue
+		}
+
 		err := ifc.LUID.DeleteIPAddress(*a)
 		if err != nil {
-			erracc = fmt.Errorf("deleting IP %q: %w", *a, err)
+			erracc = append(erracc, fmt.Errorf("deleting IP %q: %w", *a, err))
 		}
 	}
 
 	for _, a := range add {
 		err := ifc.LUID.AddIPAddress(*a)
 		if err != nil {
-			erracc = fmt.Errorf("adding IP %q: %w", *a, err)
+			erracc = append(erracc, fmt.Errorf("adding IP %q: %w", *a, err))
+		}
+	}
+
+	for _, a := range ll {
+		mib, err := ifc.LUID.IPAddress(a.IP)
+		if err != nil {
+			erracc = append(erracc, fmt.Errorf("setting skip-as-source on IP %q: unable to retrieve MIB: %w", *a, err))
+			continue
+		}
+		mib.SkipAsSource = true
```

Comment:
and if they already have SkipAsSource set? can we avoid this Set call?

Reply:
Done